magda-web-client/src/UI/DataPreviewMap.js

```diff
@@ -1,18 +1,88 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import './DataPreviewMap.css';
+import { config } from '../config'
+
+const defaultDataSourcePreference = ['GeoJSON','WFS','WMS','csv-geo-au','KML'];
+
+function determineDistribution(props)
+{
+    const distributions = props.dataset.distributions;
+    let dataSourcePreference = props.dataset.dataSourcePreference;
```

Comment:
We can also use the broken link sleuther here to check if the link if broken first? Otherwise even if the distribution meets the requirement we still can't visualise it on map. The field is called `linkActive(boolean)` and
`linkStatusAvailable(boolean)`

Reply:
@chloeleichen 
I think it's a good idea.
Will add a new commit for this. Thanks for letting me know.